api/src/main/scala/za/co/absa/loginsvc/rest/service/jwt/JWTService.scala

```diff
@@ -140,13 +143,24 @@ class JWTService @Autowired()(jwtConfigProvider: JwtConfigProvider, authSearchSe
     (refreshedAccessToken, refreshToken)
   }
 
-  def publicKey: PublicKey = keyPair.getPublic
+  def publicKey: (PublicKey, Option[PublicKey]) = {
+    val currentPublicKey = keyPair._1.getPublic
+    val previousPublicKey = keyPair._2.map(_.getPublic)
+    (currentPublicKey, previousPublicKey)
```

Comment:
I think it is a good rule of thumb to avoid calling `._1` and `._2` on a tuple -- it is advisable to rather give the fields a name for better readablity.

```suggestion
    val currentPublicKey = keyPair._1.getPublic
    val previousPublicKey = keyPair._2.map(_.getPublic)
    (currentPublicKey, previousPublicKey)
```

Ideally, you may consider applying this rule elsewhere as well.